src/OpenLoco/src/Map/Tile.cpp

```diff
@@ -144,15 +144,26 @@ namespace OpenLoco::World
                 result = tile.as<StationElement>();
                 if (result != nullptr)
                 {
-                    break;
+                    return result;
                 }
             }
             auto* elRoad = tile.as<RoadElement>();
             if (elRoad == nullptr)
```

Comment:
In vanilla it would just loop until it found any station element I tried to improve this and add some validation but made a mistake as road elements layered ontop of each other will of course have different rotations and roadIds. The only constant between them is the base height. So the extra valdiation should only have been on the baseZ (And if it was a non road element).